pyvrp/Population.py

```diff
@@ -30,23 +24,19 @@ class Population:
 
     def __init__(
         self,
-        data: ProblemData,
-        penalty_manager: PenaltyManager,
-        rng: XorShift128,
+        initial_solutions: List[Individual],
```

Comment:
Can we make the type here a general iterable? It doesn't necessarily have to be a list.

Reply:
I think I'd make sense for `initial_solutions` to be optional. By default we construct an empty population. This'd be possible if we did, for example,
```python
def __init__(
  self,
  diversity_op,
  initial_solutions: Iterable[Individual] = tuple(),
  params: PopulationParams = PopulationParams()
)
```

Reply:
I don't think that making the initial solution argument is desirable. Creating an empty Population is not a good default, and it also doesn't work well with the restarting mechanism.

Reply:
>Creating an empty Population is not a good default

Why not? That Population should not be empty is a problem for the GeneticAlgorithm, not for the Population container itself.

>and it also doesn't work well with the restarting mechanism.

This I don't understand. Restart would just return to an empty population again; I don't see what'd go wrong.

Reply:
Thinking about this, we should probably enforce in the GA that the population has least 1/2 (I think 2?) individuals, right now it can be empty.

Reply:
I'll add a PR for that.